src/main/java/edu/ie3/datamodel/utils/ContainerUtils.java

```diff
@@ -524,8 +525,10 @@ public static SubGridContainer withTrafoNodeAsSlack(final SubGridContainer subGr
         subGridContainer.getRawGrid().getTransformer2Ws().stream()
             .map(
                 oldTrafo2w ->
-                    copyTrafo2WUpdateNodeA(
-                        oldTrafo2w, oldToNewTrafo2WANodes.get(oldTrafo2w.getNodeA())))
+                    oldTrafo2w
+                        .copy()
+                        .nodeA(oldToNewTrafo2WANodes.get(oldTrafo2w.getNodeA()))
```

Comment:
A lot of "old" here. I would suggest to get the new node and assign it to a variable to have it a bit more explicitly.